src/hooks/CustomTaskHooks.ts

```diff
@@ -1,24 +1,29 @@
-import { useState } from 'react';
-import uuid from 'uuidv4';
 import { TaskType, UpdateTaskParams } from '../components/ToDoList/types';
+import { useAppDispatch, useAppSelector } from '../store/hooks';
+import {
+    addTodoAsync,
+    deleteTodoAsync,
+    updateTodoAsync,
+} from '../store/TodosSlicer';
+import { selectAllTodos } from '../store/store';
+import uuid from 'uuidv4';
 
-function useCustomTaskHooks(todoTasks: Array<TaskType>) {
-    const [tasks, setTasks] = useState<Array<TaskType>>(todoTasks);
+function useCustomTaskHooks() {
+    const dispatch = useAppDispatch();
+    const todos: Array<TaskType> = useAppSelector(selectAllTodos);
 
     const addTask = (task: string) => {
-        setTasks([
-            ...tasks,
-            {
-                id: uuid(),
-                description: task,
-                isCompleted: false,
-                isEditing: false,
-            },
-        ]);
+        const todo = {
+            id: uuid(),
```

Comment:
Usually the backend service generates the ID. Transmit without ID, and generate it at the service level